### PR TITLE
feat: use rpc for forcing the redeploy of an ecs service

### PR DESCRIPTION
**Pull request by yrobla**
Remove the column to force the new deployment and instead convert it into an RPC call.

Signed-off-by: Yolanda Robla <yolanda@iasql.com>

### CODE REVIEW

File: src/modules/0.0.23/aws_ecs_simplified/migration/1665483070412-aws_ecs_simplified.ts [hunk 1]
```diff
@@ -1,35 +0,0 @@
-import { MigrationInterface, QueryRunner } from 'typeorm';
```
Comment (dfellis) [inline src/modules/0.0.23/aws_ecs_simplified/migration/1665483070412-aws_ecs_simplified.ts:1]: Why is this file completely deleted?

Comment (dfellis) [reply src/modules/0.0.23/aws_ecs_simplified/migration/1665483070412-aws_ecs_simplified.ts:1]: Oh, wait. Is it even used?

Comment (yrobla) [reply src/modules/0.0.23/aws_ecs_simplified/migration/1665483070412-aws_ecs_simplified.ts:1]: looks like not

File: examples/ecs-fargate/prisma/infra/update.js [hunk 2]
```diff
@@ -28,11 +28,14 @@ async function main() {
...
+  })).arn.toString();
+
   console.log('Force new deployment')
-  await prisma.ecs_simplified.update({
-    where: { name: APP_NAME },
-    data: { force_new_deployment: true }
-  });
+  await prisma.$queryRaw`SELECT * from deploy_service('${arn}');`;
+
   const commit = await prisma.$queryRaw`SELECT * from iasql_commit();`
```
Comment (aguillenv) [inline examples/ecs-fargate/prisma/infra/update.js:39]: is this commit necessary?

File: src/ormconfig.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+const { SnakeNamingStrategy } = require('typeorm-naming-strategies');
```
Comment (aguillenv) [inline src/ormconfig.js:1]: this file should not be commited

File: examples/ecs-fargate/prisma/infra/update.js (hunk 2 above)
Comment (mtp1376) [reply examples/ecs-fargate/prisma/infra/update.js:39]: I also think this might not be necessary.

File: site/docs/fargate-tutorials/django.md [hunk 4]
```diff
@@ -38,34 +39,20 @@ Make sure to copy the PostgreSQL connection string as you will not see it again.
...
- aws_elb                  | target_group                  |            0
- aws_elb                  | listener                      |            0
- aws_elb                  | load_balancer_security_groups |            0
- aws_ecs_fargate          | cluster                       |            0
- aws_ecs_fargate          | service                       |            0
- aws_ecs_fargate          | task_definition               |            0
- aws_ecs_fargate          | container_definition          |            0
- aws_ecs_fargate          | service_security_groups       |            0
-(17 rows)
+aws_cloudwatch | log_group | 0 aws_ecr | public_repository | 0 aws_ecr | repository | 1 aws_ecr | repository_policy | 0 aws_security_group | security_group | 2 aws_security_group | security_group_rule | 0 aws_vpc | vpc | 1 aws_vpc | subnet | 3 aws_elb | load_balancer | 0 aws_elb | target_group | 0 aws_elb | listener | 0 aws_elb | load_balancer_security_groups | 0 aws_ecs_fargate |
```
Comment (mtp1376) [inline site/docs/fargate-tutorials/django.md:53]: I think there's a problem with line breaks here.

File: src/modules/0.0.23/aws_ecs_fargate/mappers/service.ts [hunk 5]
```diff
@@ -33,8 +33,7 @@ export class ServiceMapper extends MapperBase<Service> {
     (a?.securityGroups?.every(asg => !!b?.securityGroups?.find(bsg => Object.is(asg.groupId, bsg.groupId))) ??
       false) &&
     Object.is(a?.subnets?.length, b?.subnets?.length) &&
-    (a?.subnets?.every(asn => !!b?.subnets?.find(bsn => Object.is(asn, bsn))) ?? false) &&
-    Object.is(a.forceNewDeployment, b.forceNewDeployment);
+    (a?.subnets?.every(asn => !!b?.subnets?.find(bsn => Object.is(asn, bsn))) ?? false);
```
Comment (mtp1376) [inline src/modules/0.0.23/aws_ecs_fargate/mappers/service.ts:36]: I think this is only checking if `a.subnets`⊂`b.subnets` which clearly does not yield equality. You might need to check if `b.subnets` also is a subset of `a.subnets` to make sure they're equal.

File: src/modules/0.0.23/aws_ecs_fargate/rpcs/deploy_service.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    // given the service name, read the details
+    const serviceObj: Service =
+      (await this.module.service.db.read(ctx, this.module.service.generateId({ arn: arn ?? '' }))) ??
+      (await this.module.service.cloud.read(ctx, this.module.service.generateId({ arn: arn ?? '' })));
+
+    if (serviceObj) {
+      const client = (await ctx.getAwsClient(serviceObj.region)) as AWS;
+
+      const service = await this.updateService(client.ecsClient, {
```
Comment (mtp1376) [inline src/modules/0.0.23/aws_ecs_fargate/rpcs/deploy_service.ts:44]: Does `updateService` force a new deployment on the ECS service?

**Review by mtp1376 — APPROVED** [review]

File: examples/ecs-fargate/prisma/infra/update.js (hunk 2 above)
Comment (yrobla) [reply examples/ecs-fargate/prisma/infra/update.js:39]: it was there on the original code. Maybe we remove in a followup?

Comment (aguillenv) [reply examples/ecs-fargate/prisma/infra/update.js:39]: I think it is part of this change, if I'm not wrong it was applying the previous `force_new_deployment=true` change. That is not necessary anymore

Comment (yrobla) [reply examples/ecs-fargate/prisma/infra/update.js:39]: ah you're right, as the rpc does not need the commit

File: site/docs/fargate-tutorials/django.md (hunk 4 above)
Comment (yrobla) [reply site/docs/fargate-tutorials/django.md:53]: you're right, i'll disable prettier for this

File: src/modules/0.0.23/aws_ecs_fargate/mappers/service.ts (hunk 5 above)
Comment (yrobla) [reply src/modules/0.0.23/aws_ecs_fargate/mappers/service.ts:36]: i just copied the condition there... but we can do it as a followup, to isolate concepts of the commits

File: src/modules/0.0.23/aws_ecs_fargate/rpcs/deploy_service.ts (hunk 6 above)
Comment (yrobla) [reply src/modules/0.0.23/aws_ecs_fargate/rpcs/deploy_service.ts:44]: this is was older code was doing. However, there is a "forceNewDeployment" flag that i will add now, that will ensure to force it, i think it makes sense

**Review by aguillenv — APPROVED** [review]

**Review by aguillenv — APPROVED** [review]
Cannot be merged until it's pair in iasql-on-iasql is merged

**Pull request merged**